torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -450,8 +454,12 @@ void Reducer::push_rebuilt_params_for_all_indices() {
     for (size_t variable_index = 0; variable_index < variable_count;
          ++variable_index) {
       const auto index = VariableIndex{
+#ifdef _MSC_VER
```

Comment:
Shall we add ctors for `VariableIndex` and `VariableLocator` to avoid the ifdef on every invocation?